src/observations/observation_night.jl

```diff
@@ -0,0 +1,182 @@
+@doc raw"""
+    ObservationNight{T <: AbstractFloat}
+
+A set of optical observations taken by the same observatory on the same night.
+
+# Fields
+
+- `radec::Vector{RadecMPC{T}}`: vector of observations.
+- `observatory::ObservatoryMPC{T}`: observing station.
+- `night::TimeOfDay`: night of observation.
+- `date::DateTime`: mean date of observation.
+- `α::T`: mean right ascension.
+- `δ::T`: mean declination.
+- `v_α::T`: mean right ascension velocity.
+- `v_δ::T`: mean declination velocity.
+- `mag::T`: mean apparent magnitude.
+- `nobs::Int`: number of observations.
+- `idxs::Vector{Int}`: indices of original `radec` that entered the night.
+"""
+@auto_hash_equals struct ObservationNight{T <: AbstractFloat}
```

Comment:
After giving it some thought, I think what this `struct` represents, matches very well with what the notion of "tracklet" means in NEO literature (i.e., the group of observations for a given object, during a single night, from a single observatory), so I suggest to rename this `struct` to `Tracklet`.

Reply:
Another possibility is to follow Milani's `VeryShortArc` terminology, although my impression is that `Tracklet` is more common these days, so I'd be more inclined to the latter...

Reply:
I agree that both names are better than `ObservationNight`; since `Tracklet` is shorter, I've changed all corresponding names to `Tracklet`.